OsmAnd/src/net/osmand/plus/routing/data/AnnounceTimeDistances.java

```diff
@@ -174,28 +174,35 @@ public boolean isTurnStateNotPassed(float currentSpeed, double dist, int turnTyp
 	}
 
 	private boolean isDistanceLess(float currentSpeed, double dist, double leadDist) {
-		return isDistanceLess(currentSpeed, dist, leadDist, (float) leadDist / DEFAULT_SPEED);
+		return isDistanceLess(currentSpeed, dist, leadDist, 0.0f);
 	}
 
 	private boolean isDistanceLess(float currentSpeed, double dist, double leadDist, float leadTime) {
+		if (leadTime != 0.0f) {    //identifies TURN_NOW case
+			// Issue #17376: low speed adjustment for TURN_NOW timing:
+			if (currentSpeed > 0 && currentSpeed < DEFAULT_SPEED) {
+				float scaleDown = (float) Math.sqrt(currentSpeed / DEFAULT_SPEED);
+				leadDist = Math.max(POSITIONING_TOLERANCE, scaleDown * leadDist);
```

Comment:
Changing leadDist & leadTime bad practice cause these parameters are not working as intented

Reply:
Now they do, I timed it. The problem why they did not behave ok was we had floored _AnnounceTimeDistances.getSpeed()_ with DEFAULT_SPEED.